### PR TITLE
extend property types

### LINKED ISSUE #2013: extend device property types
**What would you like to be added/modified**:
Extend property types in device model, including Double, Float, Boolean and Bytes.

**Why is this needed**:
Currently,  property types only support "int" and "string". It is not enough to describe basic types like float and bool.

### PR DESCRIPTION
**Pull request by luogangyi**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind test
> /kind failing-test
/kind feature

**What this PR does / why we need it**:
Currently, property types only support "int" and "string". It is not enough to describe basic types like float and bool.

**Which issue(s) this PR fixes**:

Fixes #2013

**Special notes for your reviewer**:
None

**Does this PR introduce a user-facing change?**:

```release-note
* support using boolean, float, double and bytes to describe type of property in device model.
```

### CODE REVIEW

File: cloud/pkg/devicecontroller/controller/downstream.go [hunk 1]
```diff
@@ -270,6 +274,27 @@ func addDeviceModelAndVisitors(deviceModel *v1alpha2.DeviceModel, deviceProfile
 			property.AccessMode = string(ppt.Type.String.AccessMode)
 			property.DataType = DataTypeString
 			property.DefaultValue = ppt.Type.String.DefaultValue
+		} else if ppt.Type.Double != nil {
+			property.AccessMode = string(ppt.Type.Int.AccessMode)
+			property.DataType = DataTypeDouble
+			property.DefaultValue = ppt.Type.Int.DefaultValue
+			property.Maximum = ppt.Type.Int.Maximum
+			property.Minimum = ppt.Type.Int.Minimum
```
Comment (fisherxu) [inline cloud/pkg/devicecontroller/controller/downstream.go:282]: Double

File: cloud/pkg/devicecontroller/controller/downstream.go [hunk 2]
```diff
@@ -270,6 +274,27 @@ func addDeviceModelAndVisitors(deviceModel *v1alpha2.DeviceModel, deviceProfile
...
+			property.Unit = ppt.Type.Int.Unit
+		} else if ppt.Type.Float != nil {
+			property.AccessMode = string(ppt.Type.Int.AccessMode)
+			property.DataType = DataTypeFloat
+			property.DefaultValue = ppt.Type.Int.DefaultValue
+			property.Maximum = ppt.Type.Int.Maximum
+			property.Minimum = ppt.Type.Int.Minimum
+			property.Unit = ppt.Type.Int.Unit
```
Comment (fisherxu) [inline cloud/pkg/devicecontroller/controller/downstream.go:290]: Float

File: cloud/pkg/devicecontroller/controller/downstream.go (hunk 1 above)
Comment (luogangyi) [reply cloud/pkg/devicecontroller/controller/downstream.go]: done

File: cloud/pkg/devicecontroller/controller/downstream.go (hunk 2 above)
Comment (luogangyi) [reply cloud/pkg/devicecontroller/controller/downstream.go]: done

**Pull request merged**